sotodlib/tod_ops/fft_ops.py

```diff
@@ -141,33 +193,35 @@ def build_rfft_object(n_det, n, direction="FFTW_FORWARD", **kwargs):
 
     Returns:
 
-        a: array for the real valued side of the fft
-
-        b: array for the the complex side of the fft
-
-        t_fun: function for performing FFT (two are returned if direction=='BOTH')
+        rfft_obj: An instance of RFFTObj. See that class for details.
+                  If direction was FFTW_FORWARD rfft_obj.t_backward is None.
+                  If direction was FFTW_BACKWARD rfft_obj.forward is None.
     """
     fftargs = {"threads": _get_num_threads(), "flags": ["FFTW_ESTIMATE"]}
     fftargs.update(kwargs)
 
     a = pyfftw.empty_aligned((n_det, n), dtype="float32")
     b = pyfftw.empty_aligned((n_det, (n + 2) // 2), dtype="complex64")
     
+    t_forward = None
+    t_backward = None
     if direction == "FFTW_FORWARD":
-        t_fun = pyfftw.FFTW(a, b, direction=direction, **fftargs)
+        t_forward = pyfftw.FFTW(a, b, direction=direction, **fftargs)
     elif direction == "FFTW_BACKWARD":
-        t_fun = pyfftw.FFTW(b, a, direction=direction, **fftargs)
+        t_backward = pyfftw.FFTW(b, a, direction=direction, **fftargs)
     elif direction == "BOTH":
-        t_1 = pyfftw.FFTW(a, b, direction="FFTW_FORWARD", **fftargs)
-        t_2 = pyfftw.FFTW(b, a, direction="FFTW_BACKWARD", **fftargs)
-        return a, b, t_1, t_2
+        t_forward = pyfftw.FFTW(a, b, direction="FFTW_FORWARD", **fftargs)
+        t_backward = pyfftw.FFTW(b, a, direction="FFTW_BACKWARD", **fftargs)
     else:
         raise ValueError("direction must be FFTW_FORWARD or FFTW_BACKWARD")
 
-    return a, b, t_fun
+    rfft_obj = RFFTObj(n_det, n, a, b, t_forward, t_backward)
+
+    return rfft_obj 
 
 
-def find_inferior_integer(target, primes=[2, 3, 5, 7, 11, 13]):
+@lru_cache
+def find_inferior_integer(target, primes=(2, 3, 5, 7, 11, 13)):
```

Comment:
Can you cache the outermost calls only?  It probably doesn't matter much, but this function is recursive and the value of caching all the intermediate calculations is much less important.

That would entail creating private _find_inferior_integer, which is not cache-decorated.

Reply:
Sure